pliers/extractors/audio.py

```diff
@@ -119,4 +124,259 @@ def _extract(self, stim):
             values.append(mean_amplitude)
 
         return ExtractorResult(values, stim, self, features=['mean_amplitude'],
-                                onsets=onsets, durations=durations)
+                               onsets=onsets, durations=durations)
+
+
+class LibrosaFeatureExtractor(AudioExtractor):
+
+    ''' A generic class for audio extractors using the librosa library. '''
+
+    _log_attributes = ('hop_length', 'librosa_kwargs')
+
+    def __init__(self, feature=None, hop_length=512, **librosa_kwargs):
+        verify_dependencies(['librosa'])
+        if feature:
+            self._feature = feature
+        self.hop_length = hop_length
+        self.librosa_kwargs = librosa_kwargs
+        super(LibrosaFeatureExtractor, self).__init__()
+
+    def _get_feature_names(self):
```

Comment:
This seems potentially useful to users, so maybe we should drop the leading underscore and expose it publicly. Might actually be a good convention to have for other extractors that have a fixed set of features (I don't think it's worth going back and updating them all, but at least something to keep in mind going forward).

Reply:
Sure